lib/src/over_react_test/prop_type_utils.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:html';
+import 'dart:js';
+
+import 'package:matcher/matcher.dart';
+import 'package:meta/meta.dart';
+import 'package:over_react/component_base.dart';
+import 'package:over_react_test/jacket.dart';
+import 'package:react/react_client/react_interop.dart' show PropTypes, ReactElement;
+import 'package:test/test.dart';
+
+/// Tests the prop types of a component by calling the props, adding children if necessary, and checking the console
+/// for expected log output.
+///
+/// By default the test will validate that the console error count matches an
+/// inferred expected count (inferred via the length of
+/// customErrorMessageList and whether or not the `render()` will throw).
+/// This is useful for simple tests, but may be turned off for complex
+/// componentry tests where the console may log unrelated messages.
+///
+/// If a Component's `render()` will throw because it is relying on a prop
+/// that was not passed in correctly, that can be caught by setting
+/// `willThrow` to true and passing in the expected Error Matcher.
```

Comment:
I feel like most of the time, the consumer won't care whether it throws, or what it throws. If I were writing a propType validator, I would consider a component instance with invalid props to have undefined behavior, and wouldn't want to have to figure out what exactly could go wrong in those cases.